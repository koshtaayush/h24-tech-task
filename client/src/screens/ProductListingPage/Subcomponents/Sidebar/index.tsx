//Library
import React from 'react'
import styled from 'styled-components'

//Shared Interface
import { Category } from './../../../../typings/types'

interface Props {
    categories: Array<Category>
}

const Sidebar: React.FC<Props> = (props) => {
    
    //Props
    const { categories } = props

    return (
        <SidebarContainer>
            <h3>Kategorien</h3>
            {categories.length ? (
                <CategoryList>
                    {categories[0].childrenCategories.map(
                        ({ name, urlPath }) => {
                            return (
                                <CategoryListElement key={name} >
                                    <CategoryLink href={`/${urlPath}`}>
                                        {name}
                                    </CategoryLink>
                                </CategoryListElement>
                            )
                        }
                    )}
                </CategoryList>
            ) : (
                'Loading...'
            )}
        </SidebarContainer>
    )
}

export default Sidebar

const SidebarContainer = styled.section`
    border-right: 0.0625rem solid #eaeaec;
    margin-top: -1rem;
`

const CategoryList = styled.ul`
    list-style-type: none;
    margin: 0;
    padding: 0;
`

const CategoryListElement = styled.div`
    margin: 0 0 0 0.5rem;
    padding: 0.5rem 0;

    display: block;
    width: 95%;
    white-space: nowrap;
    cursor: pointer;
    overflow: hidden;
    text-overflow: ellipsis;
    min-height: 1.0625rem;
    color: #282c3f;
`

const CategoryLink = styled.a`
    display: block;
    white-space: nowrap;
    cursor: pointer;
    overflow: hidden;
    text-overflow: ellipsis;
    min-height: 1.0625rem;
    color: #282c3f;
    text-decoration: none;
`