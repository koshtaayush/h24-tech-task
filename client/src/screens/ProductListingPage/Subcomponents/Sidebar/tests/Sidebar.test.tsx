import React from 'react'
import { shallow } from 'enzyme'

import Sidebar from './../index'

describe('Sidebar', () => {
    it('Should match the snapshot', () => {
        const categories = [
            {
              "name": "Möbel",
              "articleCount": 68789,
              "childrenCategories": [
                {
                  "name": "Wohnzimmer",
                  "urlPath": "kategorie/wohnzimmermoebel/"
                },
                {
                  "name": "Schlafzimmer",
                  "urlPath": "kategorie/schlafzimmermoebel/"
                },
                {
                  "name": "Esszimmer",
                  "urlPath": "kategorie/esszimmermoebel/"
                },
                {
                  "name": "Büro",
                  "urlPath": "kategorie/bueromoebel/"
                },
                {
                  "name": "Badmöbel",
                  "urlPath": "kategorie/badezimmermoebel/"
                },
                {
                  "name": "Flurmöbel & Dielenmöbel",
                  "urlPath": "kategorie/flur-und-diele/"
                },
                {
                  "name": "Küche & Bar",
                  "urlPath": "kategorie/kuechenmoebel/"
                },
                {
                  "name": "Kinderzimmer",
                  "urlPath": "kategorie/kinderzimmermoebel/"
                }
              ],
              "categoryArticles": {
                "articles": [
                  {
                    "name": "Premium Komfortmatratze Smood",
                    "variantName": "180 x 200cm",
                    "prices": {
                      "currency": "EUR",
                      "regular": {
                        "value": 56999
                      }
                    },
                    "images": [
                      {
                        "path": "https://cdn1.home24.net/images/media/catalog/product/200x200/png/m/a/matratzenbezug-smood-webstoff-180-x-200cm-3477221.webp"
                      }
                    ]
                  },
                  {
                    "name": "Boxspringbett Kinx",
                    "variantName": "Stoff KINX: Grau - 180 x 200cm - 130 cm",
                    "prices": {
                      "currency": "EUR",
                      "regular": {
                        "value": 164999
                      }
                    },
                    "images": [
                      {
                        "path": "https://cdn1.home24.net/images/media/catalog/product/200x200/png/-/1/-1000062031-210726-10044100341-IMAGE-P000000001000062031.webp"
                      }
                    ]
                  }
                ]
              },
              "extensions": {
                "lexicon": {
                  "Mu": {},
                  "Words": {}
                }
              }
            }
          ]

        const component = shallow(<Sidebar categories={categories} />)

        expect(component).toMatchSnapshot()
    })
})
