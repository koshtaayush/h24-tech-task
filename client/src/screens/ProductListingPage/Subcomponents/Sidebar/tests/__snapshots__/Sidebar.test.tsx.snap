// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar Should match the snapshot 1`] = `
<styled.section>
  <h3>
    Kategorien
  </h3>
  <styled.ul>
    <styled.div
      key="Wohnzimmer"
    >
      <styled.a
        href="/kategorie/wohnzimmermoebel/"
      >
        Wohnzimmer
      </styled.a>
    </styled.div>
    <styled.div
      key="Schlafzimmer"
    >
      <styled.a
        href="/kategorie/schlafzimmermoebel/"
      >
        Schlafzimmer
      </styled.a>
    </styled.div>
    <styled.div
      key="Esszimmer"
    >
      <styled.a
        href="/kategorie/esszimmermoebel/"
      >
        Esszimmer
      </styled.a>
    </styled.div>
    <styled.div
      key="Büro"
    >
      <styled.a
        href="/kategorie/bueromoebel/"
      >
        Büro
      </styled.a>
    </styled.div>
    <styled.div
      key="Badmöbel"
    >
      <styled.a
        href="/kategorie/badezimmermoebel/"
      >
        Badmöbel
      </styled.a>
    </styled.div>
    <styled.div
      key="Flurmöbel & Dielenmöbel"
    >
      <styled.a
        href="/kategorie/flur-und-diele/"
      >
        Flurmöbel & Dielenmöbel
      </styled.a>
    </styled.div>
    <styled.div
      key="Küche & Bar"
    >
      <styled.a
        href="/kategorie/kuechenmoebel/"
      >
        Küche & Bar
      </styled.a>
    </styled.div>
    <styled.div
      key="Kinderzimmer"
    >
      <styled.a
        href="/kategorie/kinderzimmermoebel/"
      >
        Kinderzimmer
      </styled.a>
    </styled.div>
  </styled.ul>
</styled.section>
`;
